### PR TITLE
[14.0][MIG] l10n_th_tax_report

### PR DESCRIPTION
**Pull request by kittiu**
Dependencies

- [X] date_range
- [x] report_xlsx https://github.com/OCA/reporting-engine/pull/443
- [x] report_xlsx_helper https://github.com/OCA/reporting-engine/pull/456
- [x] l10n_th_partner
- [x] l10n_th_tax_invoice https://github.com/OCA/l10n-thailand/pull/188

### CODE REVIEW

**Comment by kittiu** [comment]
Only export to PDF is working now.

image

**Comment by kittiu** [comment]
@ps-tubtim @Saran440 @newtratip if you go chance to do this kind of report. Please help.

File: l10n_th_tax_report/reports/tax_report.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    def _get_html(self):
+        result = {}
+        rcontext = {}
+        context = dict(self.env.context)
+        report = self.browse(context.get("active_id"))
+        if report:
+            rcontext["o"] = report
+            result["html"] = self.env.ref(
+                "l10n_th_tax_report.report_tax_report_html"
+            ).render(rcontext)
```
Comment (Saran440) [inline l10n_th_tax_report/reports/tax_report.py]:
```suggestion
            )._render(rcontext)
```


In core change to _render (view)

File: l10n_th_tax_report/reports/tax_report_xlsx.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
+# Copyright 2019 Ecosoft Co., Ltd (https://ecosoft.co.th)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+import logging
+
+from odoo import models
+
```
Comment (Saran440) [inline l10n_th_tax_report/reports/tax_report_xlsx.py:7]:
```suggestion
from odoo.addons.report_xlsx_helper.report.report_xlsx_format import (
    FORMATS,
    XLS_HEADERS,
)
```

add global file format

File: l10n_th_tax_report/reports/tax_report_xlsx.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        elif objects.tax_id.type_tax_use == "purchase":
+            ws_params["ws_name"] = "Purchase TAX Report"
+            ws_params["title"] = "Purchase TAX Report"
+
+        return [ws_params]
+
+    def _vat_report(self, wb, ws, ws_params, data, objects):
+        ws.set_portrait()
+        ws.fit_to_pages(1, 0)
+        ws.set_header(self.xls_headers["standard"])
```
Comment (Saran440) [inline l10n_th_tax_report/reports/tax_report_xlsx.py]:
```suggestion
        ws.set_header(XLS_HEADERS["xls_headers"]["standard"])
```

File: l10n_th_tax_report/reports/tax_report_xlsx.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        ws.set_footer(self.xls_footers["standard"])
```
Comment (Saran440) [inline l10n_th_tax_report/reports/tax_report_xlsx.py]:
```suggestion
        ws.set_footer(XLS_HEADERS["xls_footers"]["standard"])
```

File: l10n_th_tax_report/reports/tax_report_xlsx.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        self._set_column_width(ws, ws_params)
+        row_pos = 0
+        # title
+        row_pos = self._write_ws_title(ws, row_pos, ws_params, True)
+        # company data
+        ws.write_column(row_pos, 1, ["Period :", "Partner :"], self.format_left_bold)
```
Comment (Saran440) [inline l10n_th_tax_report/reports/tax_report_xlsx.py]:
```suggestion
        ws.write_column(row_pos, 1, ["Period :", "Partner :"], FORMATS["format_left_bold"])
```

change all `self.format_xxxx` to `FORMATS["format_xxxx"]`

**Review by Saran440 — COMMENTED** [review]
Code reviewed.

you can see example from https://github.com/OCA/l10n-thailand/pull/214

**Review by ps-tubtim — APPROVED** [review]
Functional Test OK.

**Pull request merged**